src/utils/viewer/viewer.cpp

```diff
@@ -863,11 +864,25 @@ Viewer::Viewer(const Arguments& arguments)
       .addOption("agent-transform-filepath")
       .setHelp("agent-transform-filepath",
                "Specify path to load camera transform from.")
+      .addOption("quarter-resolution-scaling-limit", "2.0")
```

Comment:
I guess a more intuitive interface to expose to the command line would be a simple bool, something like "--reduce-resolution-for-high-dpi-displays". Which should probably set the scaling limit to something like 1.6 or 1.9, right?

Reply:
Alternately, at least a better description for this arg so people know how to use it.

Reply:
That's the thing, I didn't expect people to use this option :) The default behavior makes sense to me. The option is there just as a fallback, to tune when the size reduction kicks in, in rare cases such as when the desktop scaling is configured weird, or when the GPU is not powerful enough to render at full resolution, or when making screenshots and so you _do_ want a full resolution even on a HiDPI screen.

As I understood the request, the main objective was to reduce the resolution _by default_ on very-high-DPI displays, which is for me with 200% display scale and higher. On lower density displays with 1.5x or 1.6x scaling it looks visibly uglier (as it did on my 4K 23" screen), so there it isn't enabled, while for >=2x scaled displays you'll see the difference only if you really focus on it (as it was the case on my 4K 16" display).

Reply:
Ok, I think I understand now. `quarter-resolution-scaling-limit==2.0` means "reduce resolution for very-high-DPI displays" (by which we mean Retina and other 200%+ display scales). `quarter-resolution-scaling-limit==9999` would mean "don't reduce resolution, regardless of DPI". `quarter-resolution-scaling-limit==1.0` would mean "always reduce resolution, regardless of DPI".

Anyway I don't think we particularly need this feature for the C++ viewer, which isn't used much any more. This is a helpful reference. At some point, @0mdc or I can port this to the HITL framework's replay-renderer, where Retina has been bothering us.